lib/content-services/src/lib/category/services/category-tree-datasource.service.ts

```diff
@@ -56,25 +56,29 @@ export class CategoryTreeDatasourceService extends TreeService<CategoryNode>  {
             }
             const treeResponse: TreeResponse<CategoryNode> = {entries: nodesList, pagination: response.list.pagination};
             return treeResponse;
-        })) : this.categoryService.searchCategories(name, skipCount, maxItems).pipe(map((pagingResult) => {
+        })) : this.categoryService.searchCategories(name, skipCount, maxItems).pipe(mergeMap((pagingResult) => {
             const nextAfterGeneralPathPartIndex = 3;
             const pathSeparator = '/';
-            return {
-                entries: pagingResult.list.entries.map((category) => {
-                    const path = category.entry.path.name.split(pathSeparator).slice(nextAfterGeneralPathPartIndex)
-                        .join(pathSeparator);
-                    return {
-                        id: category.entry.id,
-                        nodeName: path ? `${path}/${category.entry.name}` : category.entry.name,
-                        parentId: category.entry.parentId,
-                        level: 0,
-                        nodeType: TreeNodeType.RegularNode,
-                        hasChildren: false,
-                        isLoading: false
-                    };
-                }),
-                pagination: pagingResult.list.pagination
-            };
-        }));
+            return from(pagingResult.list.entries).pipe(mergeMap((category) => {
```

Comment:
Do you need these extra from(pagingResult.list.entries) and mergeMap? You have already mergeMap in 59th line so can't you skip these 2 things? It would look something like: 
```ts
.pipe(mergeMap((pagingResult) => {
  const nextAfterGeneralPathPartIndex = 3;
  const pathSeparator = '/';
  const path = category.entry.path.name.split(pathSeparator).slice(nextAfterGeneralPathPartIndex)
                    .join(pathSeparator);
   return this.categoryService.getCategory(category.entry.id).pipe(....)
}
....
```

GetCategory returns Observable so that observable can be used by mergeMap. Just wondering if we can simplify that in this way?